sympy/matrices/expressions/tests/test_dotproduct.py

```diff
@@ -5,11 +5,17 @@
 A = Matrix(3, 1, [1, 2, 3])
 B = Matrix(3, 1, [1, 3, 5])
 C = Matrix(4, 1, [1, 2, 4, 5])
-
+D = Matrix(2, 2, [1, 2, 3, 4])
 
 def test_docproduct():
     assert DotProduct(A, B).doit() == 22
-    raises(TypeError, lambda: DotProduct(A.T, B).doit())
-    raises(TypeError, lambda: DotProduct(A, B.T).doit())
-    raises(TypeError, lambda: DotProduct(B, C).doit())
+    assert DotProduct(A.T, B).doit() == 22
+    assert DotProduct(A, B.T).doit() == 22
     assert DotProduct(A.T, B.T).doit() == 22
+
+    raises(TypeError, lambda: DotProduct(1, A))
```

Comment:
Got it. NumPy has somewhat overloaded the term to refer to a particular kind of sum-product, which is why I was confused.